### ISSUE TITLE
Support explicit PRNG management in flax transformations

### ISSUE THREAD

**Issue by Cortys** [pwhofman/probly#191]
Flax Modules require explicit management of stochasticity, i.e., a PRNG has to be passed at init (and sometimes at call) time.

Our Dropout and DropConnect transformations should allow the user to explicitly pass in their own PRNGs for Module initialization.

A second, more challenging, aspect is whether (and how) we want to support added call-time PRNGs.

In [`flax.nn.Dropout`](https://flax.readthedocs.io/en/latest/_modules/flax/nnx/nn/stochastic.html#Dropout), a PRNG should be passed in at call-time (with a fallback to a PRNG provided at init time).

Currently, if we insert flax Dropout layers, those layers cannot be called with an explicit PRNG. There might not be an elegant solution for this; in this case inserted Dropout layers will always use the PRNG provided during initialization.
